plugins/inputs/logfile/tail/tail_test.go

```diff
@@ -99,66 +98,65 @@ func TestStopAtEOF(t *testing.T) {
 	assert.Equal(t, errStopAtEOF, tail.Err())
 
 	// Read to EOF
-	for i := 0; i < linesWrittenToFile - 3; i++ {
+	for i := 0; i < linesWrittenToFile-3; i++ {
 		<-tail.Lines
 	}
 
 	// Verify StopAtEOF() has completed.
 	select {
 	case <-done:
 		t.Log("StopAtEOF() completed (as expected)")
-	case <- time.After(time.Second * 1):
+	case <-time.After(time.Second * 1):
 		t.Fatalf("StopAtEOF() has not completed")
 	}
 
 	// Then remove the tmpfile
-	if err := os.Remove(tmpfile.Name()); err != nil {
-		t.Fatalf("failed to remove temporary log file %v: %v", tmpfile.Name(), err)
-	}
+	err := os.Remove(tmpfile.Name())
+	assert.NoError(t, err)
+
 	verifyTailerExited(t, tail)
 }
 
 func setup(t *testing.T) (*os.File, *Tail, *testLogger) {
 	tmpfile, err := ioutil.TempFile("", "example")
-	if err != nil {
-		t.Fatalf("failed to create temp file: %v", err)
-	}
+	assert.NoError(t, err)
 
 	// Write the file content
 	for i := 0; i < linesWrittenToFile; i++ {
-		if _, err := fmt.Fprintf(tmpfile, "%v some log line\n", time.Now()); err != nil {
-			log.Fatal(err)
-		}
-	}
-	if err := tmpfile.Close(); err != nil {
-		log.Fatal(err)
+		_, err := fmt.Fprintf(tmpfile, "%v some log line\n", time.Now())
+		assert.NoError(t, err)
 	}
 
+	err = tmpfile.Close()
+	assert.NoError(t, err)
+
 	// Modify the exit on deletion wait to reduce test length
 	exitOnDeletionCheckDuration = 100 * time.Millisecond
 	exitOnDeletionWaitDuration = 500 * time.Millisecond
 
 	// Setup the tail
 	var tl testLogger
-	tail, err := TailFile(tmpfile.Name(), Config{
-		Logger: &tl,
-		ReOpen: false,
-		Follow: true,
-	})
-	if err != nil {
-		t.Fatalf("failed to tail file %v: %v", tmpfile.Name(), err)
-	}
+	numUsedFds := semaphore.NewSemaphore(1)
+	tail, err := TailFile(tmpfile.Name(), numUsedFds,
+		Config{
+			Logger: &tl,
+			ReOpen: false,
+			Follow: true,
+		})
+	assert.NoError(t, err)
+
+	//Increase the tailer in queue by 1 to later confirmed if the tailer in queue has been released yet in line 181
+	tail.numUsedFds.Acquire()
+	assert.Equal(t, 1, tail.numUsedFds.GetCount())
```

Comment:
This is not useful for an assertion. This just duplicates any testing on the `Acquire` function that you already have